verilog/tools/ls/symbol-table-handler_test.cc

```diff
@@ -212,6 +213,329 @@ TEST(SymbolTableHandlerTest, DefinitionNotTrackedFile) {
+// the renamable range actually returns something useful.
+// (verify: is this #1678 fixing it ?)
+TEST(SymbolTableHandlerTest,
+     FindRenamableRangeAtCursorReturnsNullDefinitionUnknown) {
+  const auto tempdir = ::testing::TempDir();
+  const std::string sources_dir =
+      verible::file::JoinPath(tempdir, __FUNCTION__);
+  ASSERT_TRUE(verible::file::CreateDir(sources_dir).ok());
+
+  absl::string_view filelist_content = "b.sv\n";
+
+  const verible::file::testing::ScopedTestFile filelist(
+      sources_dir, filelist_content, "verible.filelist");
+  const verible::file::testing::ScopedTestFile module_a(sources_dir,
+                                                        kSampleModuleA, "a.sv");
+  const verible::file::testing::ScopedTestFile module_b(sources_dir,
+                                                        kSampleModuleB, "b.sv");
+  verible::lsp::PrepareRenameParams parameters;
+  parameters.textDocument.uri =
+      verible::lsp::PathToLSPUri(sources_dir + "/b.sv");
+  parameters.position.line = 3;  // pointing to 'vara'
+  parameters.position.character = 5;
+
+  std::filesystem::absolute({sources_dir.begin(), sources_dir.end()}).string();
+  std::shared_ptr<VerilogProject> project = std::make_shared<VerilogProject>(
+      sources_dir, std::vector<std::string>(), "");
+  SymbolTableHandler symbol_table_handler;
+  symbol_table_handler.SetProject(project);
+
+  verilog::BufferTrackerContainer parsed_buffers;
+  parsed_buffers.AddChangeListener(
+      [&symbol_table_handler, parameters](
+          const std::string &uri,
+          const verilog::BufferTracker *buffer_tracker) {
+        if (!buffer_tracker) {
+          symbol_table_handler.UpdateFileContent(
+              verible::lsp::LSPUriToPath(parameters.textDocument.uri), nullptr);
+          return;
+        }
+        if (!buffer_tracker->last_good()) return;
+        symbol_table_handler.UpdateFileContent(
+            verible::lsp::LSPUriToPath(parameters.textDocument.uri),
+            &buffer_tracker->last_good()->parser());
+      });
+
+  // Add trackers for the files we're going to process - normally done by the
+  // LSP but we don't have one
+  auto b_buffer = verible::lsp::EditTextBuffer(kSampleModuleB);
+  parsed_buffers.GetSubscriptionCallback()(parameters.textDocument.uri,
+                                           &b_buffer);
+  symbol_table_handler.BuildProjectSymbolTable();
+  ASSERT_TRUE(parsed_buffers.FindBufferTrackerOrNull(
+                  parameters.textDocument.uri) != nullptr);
+
+  std::optional<verible::lsp::Range> edit_range =
+      symbol_table_handler.FindRenameableRangeAtCursor(parameters,
+                                                       parsed_buffers);
+
+  // This was ASSERT_FALSE(), but in fact we get the range of 'vara' back,
+  // and renaming that actually works. TODO: Check intent of this test.
+  ASSERT_TRUE(edit_range.has_value()) << (nlohmann::json)*edit_range;
```

Comment:
This was testing `ASSERT_FALSE()`, but it now actually returns a range. Was that #1678 fixing it ?